internal/queue/promise.go

```diff
@@ -0,0 +1,76 @@
+// Copyright IBM Corp. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package queue
+
+import (
+	"fmt"
+	"time"
+
+	ierrors "github.com/IBM-Blockchain/bcdb-server/internal/errors"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+)
+
+type completionPromise struct {
+	resultCh chan interface{}
+	timeout  time.Duration
+}
+
+func NewCompletionPromise(timeout time.Duration) *completionPromise {
```

Comment:
exported function should return an exported object. Please export `completionPromise`. Usual suggestion is to accept interface and return object.

Reply:
Done